src/dwarf.rs

```diff
@@ -1032,9 +1032,11 @@ impl DwarfResolver {
         if let SymbolType::Variable = opts.sym_type {
             return Err(Error::new(ErrorKind::Unsupported, "Not implemented"));
         }
-        let r = self.parser.find_address(name, opts);
-        if r.is_ok() {
-            return r;
+        let elf_r = self.parser.find_address(name, opts)?;
+        if !elf_r.is_empty() {
+            // Since it is found from symtab, symtab should be
+            // complete and DWARF shouldn't provide more information.
+            return Ok(elf_r);
```

Comment:
I am not really sure I understand this change. Based on your description, the problem being fixed was that we didn't stop the search in some instances. That seems to be accomplished by the additional name check below. But here we are completely changing semantics, from continuing on error earlier on and now returning early in this case, no?

Reply:
DWARF symbol resolver is not always on.  It is enabled only if the user enables it and there is DWARF debug sections in the ELF file.  When DWARF is enabled, it will query the symtab of ELF firstby calling `self.parser.find_address()` since the symtab is more efficient.  The `self.parser` here is an ELF parser.

However, the function will return two entries for the same symbol if it doesn't return from here, one from symtab and the other from DWARF.  And, self.parser.find_address() may return an empty vector if an ELF file is stripped.  Sometimes, the symtab is stripped, but DWARF is there.  So, the change here can work better.

Does that make sense?

Reply:
That makes sense, thanks for the explanation. The case I am specifically interested in is the error case, though. Before this change, we were scanning debug information for the address if `self.parser.find_address` returned an error. Now we return early. It's not clear to me why we are changing behavior like this. Can you comment?

Reply:
I think it is a mistake in the previous change of returning multiple entries for a symbol.  Before that change, `self.parser.find_address()` returned Ok() only if something was found.  But that change returns Ok() even if nothing is found.  It returns Err() only if the parser is in the wrong state or fails to load data from the file.  However, the mistake is I didn't change this part properly.